src/Facets/DeBridgeFacet.sol

```diff
@@ -0,0 +1,139 @@
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { ILiFi } from "../Interfaces/ILiFi.sol";
+import { IDeBridgeGate } from "../Interfaces/IDeBridgeGate.sol";
+import { LibAsset } from "../Libraries/LibAsset.sol";
+import { LibSwap } from "../Libraries/LibSwap.sol";
+import { ReentrancyGuard } from "../Helpers/ReentrancyGuard.sol";
+import { SwapperV2 } from "../Helpers/SwapperV2.sol";
+import { Validatable } from "../Helpers/Validatable.sol";
+import { LibDiamond } from "../Libraries/LibDiamond.sol";
+
+/// @title DeBridge Facet
+/// @author LI.FI (https://li.fi)
+/// @notice Provides functionality for bridging through DeBridge Protocol
+contract DeBridgeFacet is ILiFi, ReentrancyGuard, SwapperV2, Validatable {
+    /// Storage ///
+
+    bytes32 internal constant NAMESPACE = keccak256("com.lifi.facets.debridge");
+
+    /// @notice The contract address of the spoke pool on the source chain.
+    IDeBridgeGate private immutable deBridgeGate;
+
+    /// Types ///
+
+    /// @param executionFee Fee paid to the transaction executor.
+    /// @param flags Flags set specific flows for call data execution.
+    /// @param fallbackAddress Receiver of the tokens if the call fails.
+    /// @param data Message/Call data to be passed to the receiver
+    ///             on the destination chain during the external call execution.
+    struct SubmissionAutoParamsTo {
+        uint256 executionFee;
+        uint256 flags;
+        bytes fallbackAddress;
+        bytes data;
+    }
+
+    /// @param permit deadline + signature for approving the spender by signature.
+    /// @param useAssetFee Use assets fee for pay protocol fix (work only for specials token)
+    /// @param nativeFee Native fee for the bridging when useAssetFee is false.
+    /// @param referralCode Referral code.
+    /// @param autoParams Structure that enables passing arbitrary messages and call data.
+    struct DeBridgeData {
+        bytes permit;
+        bool useAssetFee;
+        uint256 nativeFee;
+        uint32 referralCode;
+        SubmissionAutoParamsTo autoParams;
+    }
+
+    /// Constructor ///
+
+    /// @notice Initialize the contract.
+    /// @param _deBridgeGate The contract address of the DeBridgeGate on the source chain.
+    constructor(IDeBridgeGate _deBridgeGate) {
+        deBridgeGate = _deBridgeGate;
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via DeBridge
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _deBridgeData data specific to DeBridge
+    function startBridgeTokensViaDeBridge(ILiFi.BridgeData calldata _bridgeData, DeBridgeData calldata _deBridgeData)
+        external
+        payable
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
```

Comment:
I dont understand why the doeNotContainDestinationCalls modifier applies here.....I thought deBridge supports message transfers? If they do, then this flag will be an issue as soon as we set bridgeData.hasDestinationCalls to true